### PR TITLE
Improve documentation of rate-based models

### PR DESCRIPTION
**Pull request by ddahmen**
This pull request is related to issue #1467: Improve documentation for rate-based models.
It comprises the following changes:
1. Added equation for the rate_transformer to the documentation of the template.
2. The documentation of all rate neurons was updated by adding the following information:
    - which neurons are available (input noise ipn, output noise opn, rate transformer)  
    - how to create them (explicit mentioning of suffixes ipn and opn for input and output noise)
    - preparation for cross-links to templates
    - note on parameters

### CODE REVIEW

**Comment by ddahmen** [comment]
Suitable reviewers would be @sarakonradi and @jstapmanns

**Comment by heplesser** [comment]
@ddahmen Could you edit the issue title so it reflects the changes made and mention the issue fixed only in the PR description? We generate release notes from the PR titles, so they should be informative.

File: models/lin_rate.h [hunk 1]
```diff
@@ -42,23 +42,34 @@ Name: lin_rate - Linear rate model
...
-the gap-junction framework.
+lin_rate is an implementation of a rate model with linear input
+function \f$ input(h) = g * h \f$. It either models a rate neuron with
+input noise (see rate_neuron_ipn), a rate neuron with output noise
+(see rate_neuron_opn) or a rate transformer (see
+rate_transformer_node).
+
+Linear rate neurons support multiplicative coupling which can be
+switched on and off via the boolean parameter mult_coupling
+(default=false). In case multiplicative coupling is actived the
```
Comment (sarakonradi) [inline models/lin_rate.h]: ``In case multiplicative coupling is active, the``

File: models/lin_rate.h [hunk 2]
```diff
@@ -42,23 +42,34 @@ Name: lin_rate - Linear rate model
...
+excitatory input of the model is multiplied with the function \f$
+mult\_coupling\_ex(rate) = g_{ex} * ( \theta_{ex} - rate ) \f$ and the
+inhibitory input is multiplied with the function \f$
+mult\_coupling\_in(rate) = g_{in} * ( \theta_{in} + rate ) \f$.
+
+The model supports connections to other rate models with either zero
+or non-zero delay, and uses the secondary_event concept introduced
```
Comment (sarakonradi) [inline models/lin_rate.h]: ``and it uses the secondary_event concept``

File: models/sigmoid_rate.h [hunk 3]
```diff
@@ -46,16 +46,25 @@ Name: sigmoid_rate - rate model with sigmoidal gain function
...
 
+Nonlinear rate neurons can be created by typing
+nest.Create('sigmoid_rate_ipn'). Nonlinear rate transformers can be
+created by typing nest.Create('rate_transformer_sigmoid').
+
+
 Parameters:
 
-The following parameters can be set in the status dictionary.
+The following parameters can be set in the status dictionary.Note
```
Comment (sarakonradi) [inline models/sigmoid_rate.h]: Add space after full stop: ``dictionary. Note that``

**Comment by ddahmen** [comment]
Thanks @sarakonradi and @jstapmanns for going through my changes.
I followed your suggestions and updated the documentation accordingly.

**Pull request merged**